libyul/Utilities.cpp

```diff
@@ -91,9 +91,17 @@ u256 solidity::yul::valueOfNumberLiteral(Literal const& _literal)
 {
 	yulAssert(_literal.kind == LiteralKind::Number, "Expected number literal!");
 
-	std::string const& literalString = _literal.value.str();
```

Comment:
We could do this only for decimal numbers. Parsing a hex number is much faster than parsing a decimal number.